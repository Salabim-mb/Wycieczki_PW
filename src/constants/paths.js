export const paths = {
	DASHBOARD: {
		name: "Strona główna",
		path: "/",
	},
	BLOG: {
		name: 'Blog',
		path: '/:blog',
	},
	BLOG_CATEGORIES: {
		name: "Kategorie",
		path: "/blog/:category",
		redirect: (newCategory) => `/blog/${newCategory}`,
	},
	BLOG_ENTRY: {
		name: "Wpis",
		path: "/blog/:category/:entry_id",
		redirect: (category, entryId) => `/blog/${category}/${entryId}`,
	},
	POST_CREATOR: {
		name: 'Kreator wpisów',
		path: '/blog/post-creator/:id?',
		redirect: (entryId) => `/blog/post-creator/${entryId}`
	},
	CREATOR_CATEGORY: {
		name: "Krator kategorii",
		path: "/blog/category-creator/:id?",
		redirect: (categoryId) => `/blog/category-creator/${categoryId}`
	}
}