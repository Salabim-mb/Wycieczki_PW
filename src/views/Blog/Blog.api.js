import paths from 'constants/api'

export const getCategories = async (params) => {
    const url = `${paths.BLOG_TOPICS}?blog=${params.blog}`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
};

export const getBlog = async (params) => {
    const url = `${paths.BLOG}${params.blog}/`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
};