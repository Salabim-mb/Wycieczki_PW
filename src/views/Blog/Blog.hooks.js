import { useQuery, useMutation } from 'react-query'
import { getBlog, getCategories, deleteCategory } from './Blog.api'


export const useQueryCategories = (params) => useQuery('categories', () => getCategories(params), { retry: false })
export const useQueryBlog = (params) => useQuery('blog', () => getBlog(params), { retry: false })
export const useMutationDeleteCategory = () => useMutation('category', (id) => deleteCategory(id), { retry: false })
