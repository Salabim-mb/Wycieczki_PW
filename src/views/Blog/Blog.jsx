import React from 'react';
import { useParams, Redirect } from 'react-router-dom'
import paths from 'constants/api'
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Skeleton from '@material-ui/lab/Skeleton';
import CategoryTile from 'components/CategoryTile'
import { useQueryBlog, useQueryCategories } from './Blog.hooks'


const Post = () => {
    const params = useParams()

    const blog = useQueryBlog(params)
    const categories = useQueryCategories(params)

    return (
        <Container>
            {blog.isLoading ? (
                <Skeleton variant="rect" width={210} height={118} />
            ) : null}
            {blog.isError ? <Redirect to="/error" /> : <>
                <img src={blog?.data?.image_url} alt={blog?.data?.title} />
                <Typography variant='h3'>{blog?.data?.title}</Typography>
            </>}
            {categories.isLoading ? (
                <Skeleton animation="wave" variant="circle" width={40} height={40} />
            ) : null}
            {categories.isError ? <div>Nie można załadować kategorii</div> : categories.data?.map((category) => (
                <CategoryTile key={category.id} title={category.title} cover={`${paths.PLAIN}${category.cover_url}`} desc={category.description} link={`/${params.blog}/${category.id}`} />
            ))}
        </Container>
    );
}
export default Post