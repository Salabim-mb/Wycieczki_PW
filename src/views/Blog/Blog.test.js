import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import paths from 'constants/api';
import Blog from './Blog';
import { useQueryBlog, useQueryCategories } from './Blog.hooks';

jest.mock('views/Blog/Blog.hooks', () => ({
    useQueryBlog: jest.fn(),
    useQueryCategories: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
        id: 'abc',
    }),
}));

describe('Blog', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot - data', async () => {
        useQueryBlog.mockImplementation(() => ({
            isSuccess: true,
            data: {
                title: 1,
                id: 1,
                image_url: '/link',
                topic: 1,
                show_title: true,
                color: '#000'
            },
            isError: false,
            isLoading: false,
        }));
        useQueryCategories.mockImplementationOnce(() => ({
            isSuccess: true,
            data: [{ id: 1, name: '2' }],
            isError: false,
            isLoading: false,
        }));
        const { container } = render(
            <MemoryRouter>
                <Blog />
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - isError', async () => {
        useQueryBlog.mockImplementation(() => ({
            isSuccess: true,
            data: {},
            error: 'błąd',
            isError: true,
            isLoading: false,
        }));
        useQueryCategories.mockImplementationOnce(() => ({
            isSuccess: true,
            data: [],
            error: 'błąd',
            isError: true,
            isLoading: false,
        }));
        const { container } = render(
            <MemoryRouter>
                <Blog />
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - isLoading', async () => {
        useQueryBlog.mockImplementation(() => ({
            isSuccess: true,
            data: {},
            isError: false,
            isLoading: true,
        }));
        useQueryCategories.mockImplementationOnce(() => ({
            isSuccess: true,
            data: [],
            isError: false,
            isLoading: true,
        }));
        const { container } = render(
            <MemoryRouter>
                <Blog />
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });
});