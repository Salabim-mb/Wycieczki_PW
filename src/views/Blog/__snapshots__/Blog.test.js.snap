// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog should match snapshot - data 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <img
      alt="1"
      src="/link"
    />
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      1
    </h3>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/undefined/1"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardMedia-root"
          style="background-image: url(https://wut-vtrips.herokuapp.comundefined);"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          />
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/undefined/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Czytaj dalej
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Blog should match snapshot - isError 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div>
      Nie można załadować kategorii
    </div>
  </div>
</div>
`;

exports[`Blog should match snapshot - isLoading 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="width: 210px; height: 118px;"
    />
    <img />
    <h3
      class="MuiTypography-root MuiTypography-h3"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
      style="width: 40px; height: 40px;"
    />
  </div>
</div>
`;
